### DISCUSSION TITLE
Is the IPixelCollection valid after manipulating an image?

### DISCUSSION THREAD

**Discussion by Raildex** [dlemstra/Magick.NET#1456] [General]
I need to read a lot of single pixels of an image and occasionally I manipulate it with `.Draw`

Since `GetPixels()` is extremely slow, I considered caching the PixelCollection instead and was wondering whether the pixel collection is valid after operations that manipulate the image.
What about the result of `GetArea()` in these cases aswell?
Does the Array that `GetArea()` returns represent a "realtime" view into the image or just a "copy" or the like?

**Comment by dlemstra** [comment]
I could possible introduce something else that would give you read only access to the pixels with a span that would give faster reading. The `GetArea` method now returns a copy so you should not cache this. And you should also not cache the result of `GetPixels`.

**Reply by Raildex** [reply]
Quickly reading pixels of different images is very crucial in my case. It would be great to have a faster alternative.
